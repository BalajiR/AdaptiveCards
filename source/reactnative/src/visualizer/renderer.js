/**
 * Render the adaptive card for the given payload.
 */

import React from 'react';
import { View, Text, StyleSheet, Button, Platform,Alert, Linking } from 'react-native';
import AdaptiveCards from '../adaptive-cards'

export default class Renderer extends React.Component {
    constructor(props) {
        super(props);
        this.payload = props.payload;
        this.onModalClose = props.onModalClose;
    }

    customHostConfig = {
        fontFamily:"Helvetica",
        supportsInteractivity:true,
        fontSizes: {
            small: 12,
            default: 14,
            medium: 17,
            large: 21,
            extraLarge: 26
        }
    }

    onExecuteAction = (actionObject) => {
        if(actionObject.type === "Action.Submit"){
            Alert.alert(
                'Rendered Submit',
                JSON.stringify(actionObject.data),
                [
                    { text: "Okay", onPress: () => console.log('OK Pressed') },
                ],
                { cancelable: false }
            )
        }else if(actionObject.type === "Action.OpenUrl"){
            Linking.canOpenURL(actionObject.url).then(supported => {
                if (supported) {
                  Linking.openURL(actionObject.url);
                } else {
                  console.log("Don't know how to open URI: " + actionObject.url);
                }
              }); 
        }
        
    }

    render() {
        return (
                <View style={styles.container}>
                    <View style={styles.header}>
                        <Button title="Close" onPress={this.onModalClose} />
                        <Text style={styles.title}>Adaptive Card</Text>
                        <Button title="Close" onPress={this.onModalClose} />
                    </View>

                    <AdaptiveCards
                        payload={this.payload}
                        onExecuteAction={this.onExecuteAction}
                        hostConfig = {this.customHostConfig}
                    />
                </View>
        );
    }

}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 10,
        ...Platform.select({
            ios: {
                paddingTop: 50,
            }
        }),
    },
    header: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        marginBottom: 10
    },
    title: {
        fontSize: 20
    },
});