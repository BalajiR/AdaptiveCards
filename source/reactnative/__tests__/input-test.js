import React from 'react';
import renderer from 'react-test-renderer';


import AdaptiveCard from '../src/adaptive-cards';

let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}


 const payload = {
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "type": "AdaptiveCard",
    "version": "1.0",
    "body": [
      {
        "type": "Input.Number",
        "id": "number",
        "placeholder": "Enter a number",
        "min": 1,
        "max": 10,
        "value": 3
      }
    ],
    "actions": [
      {
        "type": "Action.Submit",
        "title": "OK"
      }
    ]
  };

  customHostConfig = {
    fontFamily: "Helvetica",
    supportsInteractivity: true,
    fontSizes: {
        small: 12,
        default: 14,
        medium: 17,
        large: 21,
        extraLarge: 26
    }
}

test('AC Renderer SnapShot', () => {
    const tree = renderer.create(
        <AdaptiveCard payload = {payload}/>
    ).toJSON();

    console.log(tree);
    expect(tree).toMatchSnapshot();
   // expect(findById(tree, 'InputText')).toBeDefined()
});
