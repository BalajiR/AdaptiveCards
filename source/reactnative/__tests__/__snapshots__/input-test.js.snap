// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AC Renderer SnapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
            },
            undefined,
            Object {
              "marginTop": 10,
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          clearButtonMode="while-editing"
          keyboardType="numeric"
          maxLength={1.7976931348623157e+308}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Enter a number"
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 5,
                "borderWidth": 1,
                "marginTop": 15,
                "padding": 5,
                "width": "100%",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "400",
              },
              Object {
                "height": 44,
              },
              Object {
                "borderColor": "lightgrey",
              },
            ]
          }
          textContentType="none"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "center",
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#1D9BF6",
                    "borderRadius": 15,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginBottom": 10,
                    "marginRight": 10,
                    "padding": 10,
                  },
                  Object {
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                    },
                    Object {
                      "fontSize": 14,
                      "fontWeight": "400",
                    },
                  ]
                }
              >
                OK
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
